.github/workflows/test_dbt_models.yaml

```diff
@@ -22,31 +34,58 @@ jobs:
         with:
           role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
 
+      - name: Restore dbt state cache
+        id: cache
+        uses: ./.github/actions/restore_dbt_cache
+        with:
+          path: ${{ env.PROJECT_DIR }}/${{ env.STATE_DIR }}
+          key: ${{ env.CACHE_KEY }}
+
       - name: Install Python requirements
         run: pip install -r scripts/requirements.run_iasworld_data_tests.txt
         working-directory: ${{ env.PROJECT_DIR }}
         shell: bash
 
       - name: Run tests
         run: |
-          python3 scripts/run_iasworld_data_tests.py \
-            --target "$TARGET" \
-            --output-dir ./qc_test_results/
+          if [[ $CACHE_HIT == 'true' ]]; then
+            # shellcheck disable=SC2086
+            python3 scripts/run_iasworld_data_tests.py \
+              --target "$TARGET" \
+              --output-dir ./qc_test_results/ \
+              $SELECT_OPTION \
```

Comment:
Hmm, that's a great question! It certainly opens us up to workflow callers providing arbitrary options, although I don't actually think callers could short circuit the command itself and run an entirely different one. I made a few different attempts to hijack the command, which appeared to confirm that GitHub escapes everything correctly as a string:

* `iasworld_pardat_class_equals_luc; echo oops; exit 1; echo` ([logs](https://github.com/ccao-data/data-architecture/actions/runs/11020740704/job/30606210015#step:7:41))
* `"iasworld_pardat_class_equals_luc"; echo "oops"; exit 1;` ([logs](https://github.com/ccao-data/data-architecture/actions/runs/11020782703/job/30606352546#step:7:41))
* `"iasworld_pardat_class_equals_luc" && echo "oops" && exit 1 &&` ([logs](https://github.com/ccao-data/data-architecture/actions/runs/11020801880/job/30606420046#step:7:41))

In each case, you can see that the workflow correctly interpreted each space-separated token as an argument to `--select` and not as its equivalent bash command or keyword. Can you think of another approach that might work for a bad actor who gained access to our workflows?

Either way, I refactored the input variable so that we always add a `--select` prefix, which should help prevent a caller from passing another option that isn't supported.